lib/vaahextendflutter/services/storage/local/hive/hive_storage_impl.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:hive/hive.dart';
+
+import '../../storage.dart';
+
+/// A class implementing Storage interface using Hive as storage backend.
+class HiveStorageImpl implements Storage {
+  final String name;
+
+  Box? _box;
+
+  HiveStorageImpl({required this.name});
+
+  @override
+  Future<void> init() async {
```

Comment:
this init has to be called from parent class which will manage both hive + FSS (check below examples)

https://github.com/webreinvent/vaahflutter/blob/ff155ef61aa92bc5f07eae36289d02e1820a6cda/lib/vaahextendflutter/services/notification/push/notification.dart#L15

https://github.com/webreinvent/vaahflutter/blob/ff155ef61aa92bc5f07eae36289d02e1820a6cda/lib/vaahextendflutter/services/notification/internal/notification.dart#L35